x-pack/plugins/security_solution/public/common/components/url_state/use_url_state.tsx

```diff
@@ -61,20 +58,18 @@ const updateTimelineAtinitialization = (
 export const useUrlStateHooks = ({
   detailName,
   indexPattern,
-  history,
   navTabs,
   pageName,
-  pathName,
-  search,
   setInitialStateFromUrl,
   updateTimeline,
   updateTimelineIsLoading,
   urlState,
 }: UrlStateContainerPropTypes) => {
   const [isInitializing, setIsInitializing] = useState(true);
   const { filterManager, savedQueries } = useKibana().services.data.query;
+  const { pathname: pathName, search } = useLocation();
```

Comment:
Here is the fix! 🎉🍾 🥳 
`pathName` and `search` were passed as props from the redux store, which gets out of sync with the real pathname in the URL. So later on in the code when we used to call `histoty.replace(outdatedPathname, ...)`. It was redirecting the user to the previous page creating the redirect loop.